substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GreyToBlackObjRefVisitor.java

```diff
@@ -101,7 +101,6 @@ public boolean visitObjectReferenceInline(Pointer objRef, int innerOffset, boole
 
             // Promote the Object if necessary, making it at least grey, and ...
             Object obj = p.toObject();
-            assert innerOffset < LayoutEncoding.getSizeFromObject(obj).rawValue();
```

Comment:
Because an object can be visited from two worker threads simultaneously. `getSizeFromObject()` would return an invalid value if another thread had installed a forwarding pointer in the object's header